src/h2load.cc

```diff
@@ -1744,42 +1743,28 @@ int main(int argc, char **argv) {
               << "cores." << std::endl;
   }
 
-  if (!config.is_rate_mode()) {
-    if (config.nreqs < config.nclients) {
-      std::cerr << "-n, -c: the number of requests must be greater than or "
-                << "equal to the concurrent clients." << std::endl;
-      exit(EXIT_FAILURE);
-    }
+  if (config.nreqs < config.nclients) {
+    std::cerr << "-n, -c: the number of requests must be greater than or "
+              << "equal to the concurrent clients." << std::endl;
+    exit(EXIT_FAILURE);
+  }
 
-    if (config.nclients < config.nthreads) {
-      std::cerr << "-c, -t: the number of client must be greater than or equal "
-                   "to the number of threads." << std::endl;
-      exit(EXIT_FAILURE);
-    }
-  } else {
+  if (config.nclients < config.nthreads) {
+    std::cerr << "-c, -t: the number of client must be greater than or equal "
+                 "to the number of threads." << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  if (config.is_rate_mode()) {
     if (config.rate < config.nthreads) {
       std::cerr << "-r, -t: the connection rate must be greater than or equal "
```

Comment:
> h2load -t6 -c360 -r6 --rate-period=1 will do create 60 clients per thread at a rate of 1 client/second per thread. --rate-period is really useful in this case

Does it really do that for value `1`? It seems weird to do that it would because rate period value defaults to 1.0 if `--rate-period` is undefined, which was my first example.

> So the discussion is whether -r option specifies rate per thread or per global.
I agree this is the real point. I think it would help to update manual for `-r` to make it clear if it applies per thread or globally.